apps/hpc-ftsadmin/src/app/components/merge-modal.tsx

```diff
@@ -0,0 +1,405 @@
+import EastIcon from '@mui/icons-material/East';
+import { Box, Modal } from '@mui/material';
+import { errors, util } from '@unocha/hpc-data';
+import { C, styled } from '@unocha/hpc-ui';
+import { Form, Formik } from 'formik';
+import * as io from 'io-ts';
+import React from 'react';
+import { useNavigate } from 'react-router';
+import { toast } from 'react-toastify';
+import tw from 'twin.macro';
+import { type LanguageKey, t } from '../../i18n';
+import { getContext, getEnv } from '../context';
+import paths from '../paths';
+import { TOAST_CONFIG, TOAST_CONFIG_ERROR } from '../utils/constants';
+import { fnCategories, fnOrganizations } from '../utils/fn-promises';
+import validateForm from '../utils/form-validation';
+import { valueToInteger } from '../utils/utils';
+
+type MergeModalProps = {
+  type: 'organization' | 'keyword';
+  load: () => void;
+};
+type OrganizationMergeModalValues = {
+  mergingEntities: util.FormObjectValue[];
+  receivingEntity: util.FormObjectValue | null;
+};
+type KeywordMergeModalValues = {
+  mergingEntities: util.FormObjectValue | null;
+  receivingEntity: util.FormObjectValue | null;
+};
+
+const ModalContainer = tw.div`
+  p-8
+  my-1 
+  flex 
+  justify-between 
+  items-center
+  bg-white
+  shadow-[rgba(50,_50,_105,_0.15)_0px_2px_5px_0px,_rgba(0,_0,_0,_0.05)_0px_1px_1px_0px]
+  w-[60vw]
+  rounded-sm
+`;
+
+const MergeContainer = tw.div`
+  p-6
+  rounded-sm
+  border-unocha-panel-border
+  border-solid
+  basis-1/2
+  grow-0
+  max-w-[50%]
+`;
+
+const StyledForm = tw(Form)`
+  w-full
+`;
+
+const StyledDiv = tw.div`
+  self-center
+`;
+
+const EndIcon = styled(EastIcon)`
+  [dir='rtl'] & {
+    transform: rotate(180deg);
+  }
+`;
+
+const isKeywordValues = (
+  values: OrganizationMergeModalValues | KeywordMergeModalValues
+): values is KeywordMergeModalValues => !Array.isArray(values.mergingEntities);
+const isOrganizationValues = (
+  values: OrganizationMergeModalValues | KeywordMergeModalValues
+): values is OrganizationMergeModalValues =>
+  Array.isArray(values.mergingEntities);
+
+const parseEntityString = (
+  value: string,
+  type: 'organization' | 'keyword',
+  lang: LanguageKey
+) => {
+  const singular = t.t(lang, (s) => s.components.mergeModal[type].singular);
+  const plural = t.t(lang, (s) => s.components.mergeModal[type].plural);
+  return value.replace('{entity}', singular).replace('{entities}', plural);
+};
+const ConfirmationText = ({
+  mergingEntities,
+  receivingEntity,
+  lang,
+}: {
+  mergingEntities: util.FormObjectValue[] | (util.FormObjectValue | null);
```

Comment:
The parentheses don't do anything here, hover to see. You could have done this too:
```suggestion
  mergingEntities:
    | OrganizationMergeModalValues['mergingEntities']
    | KeywordMergeModalValues['mergingEntities'];
```